onyo/lib/command_utils.py

```diff
@@ -0,0 +1,785 @@
+        # risk of collisions between independent checkouts of a repo.
+        raise ValueError('The length of faux serial numbers must be >= 4.')
+
+    if num < 1:
+        raise ValueError('The length of faux serial numbers must be >= 1.')
+
+    alphanum = string.ascii_letters + string.digits
+    faux_serials = set()
+    repo_faux_serials = {str(x.name).split('faux')[-1] for x in repo.assets}
+
+    while len(faux_serials) < num:
+        serial = ''.join(random.choices(alphanum, k=length))
+        if serial not in repo_faux_serials:
+            faux_serials.add(f'faux{serial}')
+
+    return faux_serials
+
+
+def get_template(repo: Repo,
+                 template_name: Union[Path, str, None] = None) -> Path:
+    if not template_name:
+        template_name = repo.get_config('onyo.new.template')
+        if template_name is None:
+            log.error("Either --template must be given or 'onyo.new.template' must be set.")
+            raise ValueError("Either --template must be given or 'onyo.new.template' must be set.")
+
+    template_name = Path(template_name)
+    for template in repo.templates:
+        if template.name == template_name.name:
+            return template
+        elif template_name.is_file() and template.samefile(template_name):
+            return template
+
+    log.error(f"Template {template_name} does not exist.")
+    raise ValueError(f"Template {template_name} does not exist.")
+
+
+def valid_asset_path_and_name_available(repo: Repo,
+                                        asset: Path,
+                                        new_assets: list[Path]) -> None:
+    if not valid_name(asset):
+        log.error(f"'{asset}' is not a valid asset name.")
+        raise ValueError(f"'{asset}' is not a valid asset name.")
+    if file := [file for file in repo.assets if asset.name == file.name]:
+        log.error(f"Filename '{asset.name}' already exists as '{file[0]}'.")
+        raise ValueError(f"Filename '{asset.name}' already exists as '{file[0]}'.")
+    elif file := [file for file in new_assets if asset.name == file.name]:
+        log.error(f"Input contains multiple '{file[0].name}'")
+        raise ValueError(f"Input contains multiple '{file[0].name}'")
+    if is_protected_path(asset):
+        log.error(f"The path is protected by onyo: '{asset}'")
+        raise ValueError(f"Input contains multiple '{file[0].name}'")
+
+
+def valid_name(asset: Union[Path, str]) -> bool:
+    asset = Path(asset)
+
+    try:
+        re.findall(r'(^[^._]+?)_([^._]+?)_([^._]+?)\.(.+)', asset.name)[0]
+    except (ValueError, IndexError):
+        log.info(f"'{asset.name}' must be in the format '<type>_<make>_<model>.<serial>'")
+        return False
+
+    return True
+
+
+def set_assets(repo: Repo,
+               paths: Iterable[Union[Path, str]],
+               values: Dict[str, Union[str, int, float]],
+               dryrun: bool,
+               rename: bool, depth: Union[int]) -> str:
```

Comment:
Did you possibly overlook to add the linebreak between these two variables, or is there a reason to except the "one line per variable" rule here? :)

Reply:
Overlooked, thx!